records/2024-05-23.scalability.short-code-implementation.md

```diff
@@ -0,0 +1,107 @@
+# Short Code Implementation
+
+Date: 2025-05-23
+
+## Status
+
+**DRAFT**.
+
+## Context
+
+Notify was awarded a General Purpose Short Code by the Canadian Telecom Regulators. This ADR goes through when an SMS will be sent through a Short Code or Long Code and the decision behind why a certain method will be chosen.
+
+## Implementation Goals
+1. Assign SMS templates to send through our generic shortcode.
+1. Assign any template to high, normal, or bulk priority automatically (SMS & Email)
+1. Design a way to categorize templates
+1. Override to force a certain template to go through a certain flow
+1. The ability for users to select “Other” and enter their own use case for Template Categories
+
+## Goals to be addressed by another ADR
+1. Figure out dynamic and static pools of numbers
+1. Decide on using AWS Pinpoint or AWS SNS
+Usage of a pool of codes/ dedicated long codes
```

Comment:
Assuming that was another numbered point here.
```suggestion
1. Usage of a pool of codes/ dedicated long codes
```

Reply:
There is also a toll free number we use to send the American numbers, which would be an exceptional usage at the moment, but maybe less exceptional if we ever move further to support international SMS sending.

Reply:
Leaving this as is, to be addressed by the CORE team ADR